angular/src/app/screens/sign-up/sign-up.component.spec.ts

```diff
@@ -1,24 +1,87 @@
-import { ComponentFixture, TestBed } from '@angular/core/testing';
-
+import { CommonModule } from '@angular/common';
+import { HttpClientModule } from '@angular/common/http';
+import { ReactiveFormsModule } from '@angular/forms';
+import { AppRoutingModule } from '../../app-routing.module';
+import { FormContainerComponent } from '../../components/form-container/form-container.component';
+import { SignUpRoutingModule } from './sign-up-routing.module';
 import { SignUpComponent } from './sign-up.component';
+import { APP_BASE_HREF } from '@angular/common';
 
-describe('SignUpComponent', () => {
-  let component: SignUpComponent;
-  let fixture: ComponentFixture<SignUpComponent>;
+import { fireEvent, render, waitFor } from '@testing-library/angular';
+import { screen } from '@testing-library/dom';
+import '@testing-library/jest-dom';
+import { SIGN_UP_FIELDS } from '../../constants/formAccount';
 
+describe('FormContainerComponent', () => {
+  const createUserMock = jest.fn();
   beforeEach(async () => {
-    await TestBed.configureTestingModule({
-      declarations: [SignUpComponent],
-    }).compileComponents();
+    render(FormContainerComponent, {
+      declarations: [SignUpComponent, FormContainerComponent],
+      imports: [
+        CommonModule,
+        SignUpRoutingModule,
+        ReactiveFormsModule,
+        HttpClientModule,
+        AppRoutingModule,
+      ],
+      componentProperties: {
+        formFields: SIGN_UP_FIELDS,
+        buttonSend: 'Sign Up',
+        handleOnSubmit: {
+          emit: createUserMock,
+        } as any,
+      },
+      providers: [{ provide: APP_BASE_HREF, useValue: '/my/app' }],
+    });
   });
 
-  beforeEach(() => {
-    fixture = TestBed.createComponent(SignUpComponent);
-    component = fixture.componentInstance;
-    fixture.detectChanges();
+  it('check all empty fields', async () => {
+    const form = screen.getByRole('form');
+    fireEvent.submit(form);
+    const errorSpans = screen.getAllByTestId('error-span');
+
+    expect(errorSpans).toHaveLength(5);
```

Comment:
I prefer to map SIGN_UP_FIELDS errorSpans and check errorSpans variable includes them 😄